### ISSUE TITLE
Feature Request: Support for PFX / P12 certificate format

### ISSUE THREAD

**Issue by sdnts** [a85/POSTMan-Chrome-Extension#2494]
As suggested by @daecabhir on a [comment](https://github.com/postmanlabs/postman-app-support/issues/1884#issuecomment-261237293) on #1884

**Comment by sdnts** [comment]
We're working on support for the PFX certificate format. Will keep this thread up-to-date with the progress.

**Comment by daecabhir** [comment]
Most excellent, thank you!

**Comment by Moffmo** [comment]
Looking forward to this feature :)

**Comment by davidmorabito** [comment]
In order to use PFX in OSX (El Capitan), I did the following:

1. Create a CRT and KEY file out of your PFX following this tutorial: https://www.markbrilman.nl/2011/08/howto-convert-a-pfx-to-a-seperate-key-crt-file/
2. In Settings, disable SSL Verification
3. In Settings, add new certificate. Add your host without protocol, and pick your CRT file AND also pick your KEY file. Provide a passphrase too.

At first I thought the certs were wrong, because I used the same certificate for both inputs as some guy suggested in another issue. So in order to make it work, I needed to specify BOTH certificates.

Hope it helps someone. Took me couple of hours.

**Comment by MikeCook9994** [comment]
@madebysid Is there an update on this feature? It's been nearly a year and it's exceptionally annoying to perform the work around.

**Comment by manufunk** [comment]
Any news? Our company would love working with Postman but without PFX or CER support it is not possible for us. All openssl transformations to crt/key failed so far by handshake errors.

**Comment by syacko** [comment]
I will add my voice to the choir, I need PFX support in order to use this tool for my ADP project.

**Comment by le-munch** [comment]
Same here. I also need this feature in enterprise environment

**Comment by pdeak** [comment]
I would like this feature too, but in the meantime I created the following Windows batch script to convert a .pfx file to it's corresponding .crt and .key files (requires OpenSSH on the path):

```
@echo off

REM Takes 2 arguments:
REM 1) The name of the pfx file (without the .pfx bit)
REM 2) The password on the pfx file

openssl pkcs12 -in %1.pfx -clcerts -nokeys -out %1.crt -passin pass:%2
openssl pkcs12 -in %1.pfx -nocerts -nodes  -out %1.key -passin pass:%2
```

Then if you put the batch file in your path, you can convert blah.pfx by doing:

```convertpfx blah password```

This will produce a blah.crt and a blah.key file.

**Comment by sam-sarmento** [comment]
I also need this feature. Any news?

**Comment by andris** [comment]
workaround works but is a bit cumbersome - Adding native pfx support would be awesome and appreciated!

**Comment by pregress** [comment]
In the browser extension this just works. Because it uses the certificate store from the OS. Why isn't this the same in the native postman app? 
A prompt that lets you choose from a list of installed certificates. 

This forces me to keep using a deprecated version of postman.

**Comment by pregress** [comment]
So based on @davefuentes link in his solution. We added this to our CI pipeline, without prompting for passwords.
Powershell:
<pre><code>
$certPath = "Path\to\pfx"
$keyFilePath = "Path\To\Save\.key"
$crtFilePath = "Path\To\Save\.crt"
$password = "PasswordForPrivateKey"

# Create key file
&$openSslPath pkcs12 -in $certPath -nocerts -out $keyFilePath -password "pass:$password" -passout "pass:$password"

# Create crt file
&$openSslPath pkcs12 -in $certPath -clcerts -nokeys -out $crtFilePath -password "pass:$password" -passout "pass:$password"
</code></pre>

**Comment by rmolabanti** [comment]
+1

**Comment by ambroiseRabier** [comment]
+1

**Comment by jorgespiessens** [comment]
+1

**Comment by brnleehng** [comment]
+1

**Comment by hansen12193** [comment]
+1

**Comment by GetafixIT** [comment]
+1

**Comment by vgaddavcg** [comment]
+1

**Comment by sanesai** [comment]
We will be glad to have this feature +1

**Comment by joelabate** [comment]
> We're working on support for the PFX certificate format. Will keep this thread up-to-date with the progress. - Nov 25, 2016 

It's incredibly absurd that this hasn't been addressed in 2 years. If you're unwilling or unable to add this in, at least have the courtesy to tell us so we can stop waiting for it and move on.

**Comment by amitchaudhary** [comment]
Any plan for adding the .pfx certificate support. It's been a while?

**Comment by cpend** [comment]
+1

**Comment by shamasis** [comment]
Hello there, we have started working towards adding this. However, to unblock anyone who is blocked on this, it is quite easy to generate key and cert from pfx bundle. 
https://medium.com/@jkewley/testing-client-certificate-authentication-to-azure-api-management-with-postman-e1cfae52fc35

**Comment by codenirvana** [comment]
We've added this on our Canary (`version: 7.1.0-canary01`) channel https://www.getpostman.com/canary

![Screenshot 2019-03-28 at 12 10 55 PM](https://user-images.githubusercontent.com/4865836/55135780-a772d680-5152-11e9-8e6d-4c4ec52ee15c.png)

Will update the thread once we release this on our stable channel.

**Comment by codenirvana** [comment]
We've added support for this in the latest Postman App (v7.0.9).

**Comment by ghost** [comment]
@codenirvana Thanks, tested and working ok. However, it refused to open my pfx file when I used a password containing special characters. I had to recreate the pfx file with a password containing only letters and numbers before it worked in Postman.
